core/txpool/legacypool/legacypool.go

```diff
@@ -683,6 +683,9 @@ func (pool *LegacyPool) validateTxBasics(tx *types.Transaction, local bool) erro
 		opts.MinTip = new(big.Int)
 	}
 	if err := txpool.ValidateTransaction(tx, pool.currentHead.Load(), pool.signer, opts); err != nil {
+		if errors.Is(err, txpool.ErrInvalidSender) && pool.chainconfig.ChainID.Cmp(tx.ChainId()) != 0 {
```

Comment:
IIUC, this log condition will only be met for EIP-155 type transactions that were signed using an incorrect chain ID (where the chain ID isn't explicitly included in the tx itself, but using the wrong chain ID results in an invalid signature). 

Should we remove the "invalid sender" check, and instead just have the chain ID check? I think that way it will apply for any tx types using the wrong chain ID, including EIP-2930, EIP-1559, etc.

Reply:
> IIUC, this log condition will only be met for EIP-155 type transactions that were signed using an incorrect chain ID (where the chain ID isn't explicitly included in the tx itself, but using the wrong chain ID results in an invalid signature).
> 
> Should we remove the "invalid sender" check, and instead just have the chain ID check? I think that way it will apply for any tx types using the wrong chain ID, including EIP-2930, EIP-1559, etc.

Sure, done.